scripts/deleteAndLoadSnapsets.js

```diff
@@ -0,0 +1,523 @@
+/*!
+Copyright 2018 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+// This script modifies the preferences database:
+// 1. Update the views records for accessing Prefs Safes and GPII Keys,
+// 2. Retrieves all the Prefs Safes of type "snapset" (prefsSafesType = "snapset") from the databsse.
+// 3. Retrieves all the GPII Keys associated with each snapset Prefs Safe so found,
+// 4. Deletes these Prefs Safes and their associated GPII Keys from the database,
+// 5. Uploads the new Prefs Safes and their GPII Keys to the database,
+// A sample command that runs this script:
+// node deleteAndLoadSnapsets.js $COUCHDBURL $STATIC_DATA_DIR $BUILD_DATA_DIR
+
+"use strict";
+
+var http = require("http"),
+    url = require("url"),
+    fs = require("fs"),
+    fluid = require("infusion");
+
+var gpii = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.dataLoader");
+
+fluid.setLogging(fluid.logLevel.INFO);
+
+// Handle command line
+if (process.argv.length < 5) {
+    fluid.log("Usage: node deleteAndLoadSnapsets.js $COUCHDB_URL $STATIC_DATA_DIR $BUILD_DATA_DIR [--justDelete]");
+    process.exit(1);
+}
+
+/**
+ * Create a set of options for data loader and a function to retreive them.
+ * The options are based on the command line parameters and a set of database
+ * constants.
+ * @param {Array} processArgv - The command line arguments.
+ * @return {Object} - The options.
+ */
+gpii.dataLoader.initOptions = function (processArgv) {
+    var dbOptions = {};
+    dbOptions.couchDbUrl = processArgv[2];
+    dbOptions.staticDataDir = processArgv[3];
+    dbOptions.buildDataDir = processArgv[4];
+    if (processArgv.length > 5 && processArgv[5] === "--justDelete") { // for debugging.
+        dbOptions.justDelete = true;
+    } else {
+        dbOptions.justDelete = false;
+    }
+
+    // Set up database specific options
+    dbOptions.viewsUrl = dbOptions.couchDbUrl + "/_design/views";
+    dbOptions.prefsSafesViewUrl = dbOptions.couchDbUrl + "/_design/views/_view/findSnapsetPrefsSafes";
+    dbOptions.gpiiKeysViewUrl = dbOptions.couchDbUrl + "/_design/views/_view/findAllGpiiKeys";
+    dbOptions.parsedCouchDbUrl = url.parse(dbOptions.couchDbUrl);
+    dbOptions.staticData = [];
+    /* dbOptions.newViews; */
```

Comment:
Can line 62, 63 be removed?

Reply:
These comments are a form of documentation, implying their existence at a later time.  It's conceptually similar to, for example, the commented-out "exec" invoker in the [Mock testing component](https://github.com/GPII/universal/blob/master/gpii/node_modules/testing/src/Mocks.js#L29).  (Although, it should be `fluid.notImplemented` in the case of Mock).

Originally, lines 62 and 63 were `dbOptions.newViews = undefined;` and `dbOptions.oldViews = "";` but their real values are set later when the information is fetched.  I could go back to the original form.